blockchain/chain.go

```diff
@@ -781,24 +828,32 @@ func (b *BlockChain) calcSequenceLock(tx *btcutil.Tx, utxoView *UtxoViewpoint,
 		case sequenceNum&RelativeLockSeconds == RelativeLockSeconds:
 			// This input requires a relative time lock expressed
 			// in seconds before it can be spent. Therefore, we
-			// need to query for the block in which this input was
-			// included within so we can compute the past median
-			// time for the block prior to the one which included
-			// this referenced output.
+			// need to query for the block prior to the one in
+			// which this input was included within so we can
+			// compute the past median time for the block prior to
+			// the one which included this referenced output.
 			// TODO: caching should be added to keep this speedy
-			var blockNode *blockNode
-			if err := b.db.View(func(dbTx database.Tx) error {
-				header, err := dbFetchHeaderByHeight(dbTx, inputHeight-1)
-				if err != nil {
-					return err
-				}
-				blockSha := header.BlockHash()
-				blockNode = newBlockNode(header, &blockSha, inputHeight)
-				return nil
-			}); err != nil {
-				return nil, err
+			inputDepth := uint32(b.bestNode.height-inputHeight) + 1
+			blockNode, err := b.ancestorNode(b.bestNode, inputDepth)
+			if err != nil {
+				return sequenceLock, err
+			}
+
+			// The call above loaded all the headers between the
+			// inclusionary block and the chain tip. In order to
+			// properly calculate the proper relative lock time, we
+			// also need to load into memory enough blocks from the
+			// MTP calculation.
+			_, err = b.ancestorNode(b.bestNode, medianTimeBlocks)
```

Comment:
I don't believe this is needed, but if it were, I think it would need to be from the returned `blockNode`.  For example, say we're at block 450k and the input is from block 420k.  The call to `ancestorNode` above will get back to 420k, but you still need to end up going back to 420k-11 = 419989 in order to calculate the median time.

However, as I first mentioned, this shouldn't be needed.  The call to `b.calcPastMedianTime` will load the extra needed block nodes correctly so long as memory has at least the block after the most recent block the calculation starts from -- in keeping with the example, that would be 420k.